mesonbuild/utils/vsenv.py

```diff
@@ -92,7 +92,7 @@ def _setup_vsenv(force: bool) -> bool:
     bat_file.write(bat_contents)
     bat_file.flush()
     bat_file.close()
-    bat_output = subprocess.check_output(bat_file.name, universal_newlines=True)
+    bat_output = subprocess.check_output(bat_file.name, universal_newlines=True, encoding=getencoding())
```

Comment:
Commit message is wrong since this has nothing to do with unittests. It's also a duplicate of an existing PR.

Reply:
commit squashed with other encoding related commits. It is related to unittests because otherwise `WindowsTests.test_vsenv_option` fails on my machine. Not sure why it is ok on CI: probably because it uses a different default locale.
It hte existing PR, I used `utf-8` as encoding, but `getencoding()` is probably better...